components/automate-cli/pkg/verifyserver/services/hardwareresourcecount/hardwareresourcecount.go

```diff
@@ -0,0 +1,201 @@
+	var check = models.Checks{}
+	if set == nodecount {
+		check = createCheck(constants.IP_ADDRESSS, true, constants.UNIQUE_SUCCESS_MESSAGE, "", "")
+	} else {
+		check = createCheck(constants.IP_ADDRESSS, false, "", constants.UNIQUE_ERROR_MESSAGE, fmt.Sprintf(constants.UNIQUE_RESOLUTION_MESSAGE, nodetype))
+	}
+	return check
+}
+
+func ValidFormat(ip string) models.Checks {
+	var check = models.Checks{}
+	if net.ParseIP(ip) == nil {
+		check = createCheck(constants.IP_ADDRESSS, false, "", constants.VALID_FORMAT_ERROR_MESSAGE, fmt.Sprintf(constants.VALID_FORMAT_RESOLUTION_MESSAGE, ip))
+	} else {
+		check = createCheck(constants.IP_ADDRESSS, true, constants.VALID_FORMAT_SUCCESS_MESSAGE, "", "")
+	}
+	return check
+}
+
+// This function will check if any of the frontend nodes(Automate and CS) are shared with any of the backend nodes(Postgres and Opensearch) and vice-versa.
+func SharedIP(nodetype, ip string, setOfCluster map[string]string) models.Checks {
+	var check = models.Checks{}
+	var opposite_cluster_type string
+
+	//This is for giving the appropriate success message.
+	if nodetype == constants.AUTOMATE || nodetype == constants.CHEF_INFRA_SERVER {
+		opposite_cluster_type = constants.BACKEND_CLUSTER
+	} else {
+		opposite_cluster_type = constants.FRONTEND_CLUSTER
+	}
+
+	if setOfCluster[ip] == "" {
+		check = createCheck(constants.IP_ADDRESSS, true, fmt.Sprintf(constants.SHARED_SUCCESS_MESSAGE, opposite_cluster_type), "", "")
+	} else {
+		check = createCheck(constants.IP_ADDRESSS, false, "", fmt.Sprintf(constants.SHARED_ERROR_MESSAGE, nodetype, setOfCluster[ip]), fmt.Sprintf(constants.SHARED_RESOLUTION_MESSAGE, nodetype, setOfCluster[ip]))
+	}
+	return check
+}
+
+// This function will check if the node count we are getting in request is fulfilling the minimum HA requirements.
+func ValidCount(minNodeCount, reqNodeCount int, nodetype string) models.Checks {
+	var check = models.Checks{}
+	if reqNodeCount >= minNodeCount {
+		check = createCheck(constants.IP_ADDRESSS, true, fmt.Sprintf(constants.VALID_COUNT_SUCCESS_MESSAGE, nodetype), "", "")
+	} else {
+		check = createCheck(constants.IP_ADDRESSS, false, "", fmt.Sprintf(constants.VALID_COUNT_ERROR_MESSAGE, nodetype), fmt.Sprintf(constants.VALID_COUNT_RESOLUTION_MESSAGE, nodetype))
+	}
+	return check
+}
+
+func createCheck(title string, passed bool, successMsg, errorMsg, resolutionMsg string) models.Checks {
+	return models.Checks{
+		Title:         title,
+		Passed:        passed,
+		SuccessMsg:    successMsg,
+		ErrorMsg:      errorMsg,
+		ResolutionMsg: resolutionMsg,
+	}
+}
+
+func (hrc *HardwareResourceCountService) GetHardwareResourceCount(req models.HardwareResourceRequest) []models.HardwareResourceResponse {
+	var response = []models.HardwareResourceResponse{}
+	is_managed_services := false
+	ch := make(chan map[string]models.HardwareResourceResponse)
+
+	//This structure holds minimum HA requirements as per Public Doc
+	var minNodeCount = models.CountPerHARequirements{
```

Comment:
You can avoid creation of this variable here and passing it run method. You can create this inside RunHardwareResourceCountCheck. This will reduce the number of arguments passed in the function